bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/config/HttpThingConfig.java

```diff
@@ -43,7 +51,26 @@ public class HttpThingConfig {
     public @Nullable String contentType = null;
 
     public boolean ignoreSSLErrors = false;
+    public boolean strictErrorHandling = false;
 
     // ArrayList is required as implementation because list may be modified later
     public ArrayList<String> headers = new ArrayList<>();
+    public String userAgent = "";
+
+    public Map<String, String> getHeaders() {
+        Map<String, String> headersMap = new HashMap<>();
+        // add user agent first, in case it is also defined in the headers, it'll be overwritten
```

Comment:
Is there a reason for not using default user-agent ("Jetty" with full version information)? #15047 for reference if we need to revisit something here.